src/App.tsx

```diff
@@ -1,95 +1,129 @@
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort((g1, g2) => {
+    switch (sortType) {
+      case 'ALPHABET':
+        return g1.localeCompare(g2);
+
+      case 'LENGTH':
+        return g1.length - g2.length;
+
+      case 'NONE':
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
+export class App extends React.Component<{}, State> {
+  state = {
+    sortType: 'NONE',
+    isReversed: false,
+    isResetActive: false,
+  };
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
+  render() {
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button', 'is-info', {
+              'is-light': this.state.sortType !== 'ALPHABET',
+            })}
+            onClick={() => this.setState({
+              sortType: 'ALPHABET',
+              isResetActive: true,
+            })}
+          >
+            Sort alphabetically
+          </button>
 
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
+          <button
+            type="button"
+            className={classNames('button', 'is-success', {
+              'is-light': this.state.sortType !== 'LENGTH',
+            })}
+            onClick={() => this.setState({
+              sortType: 'LENGTH',
+              isResetActive: true,
+            })}
+          >
+            Sort by length
+          </button>
 
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
+          <button
+            type="button"
+            className={classNames('button', 'is-warning', {
+              'is-light': !this.state.isReversed,
+            })}
+            onClick={() => this.setState((state) => {
+              if (state.sortType !== 'NONE') {
+                return {
+                  isReversed: !state.isReversed,
+                  isResetActive: true,
+                };
+              }
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+              return {
+                isReversed: !state.isReversed,
+                isResetActive: !state.isResetActive,
+              };
+            })}
+          >
+            Reverse
+          </button>
+
+          {this.state.isResetActive
+          && (
```

Comment:
```suggestion
          {this.state.isResetActive && (
```
You could also destructure isResetActive from state